chutney/engine/src/main/java/com/chutneytesting/engine/domain/execution/strategies/RetryWithTimeOutStrategy.java

```diff
@@ -106,6 +109,28 @@ public Status execute(ScenarioExecution scenarioExecution,
         return st;
     }
 
+    private long evaluateAndConvertToMilliseconds(String expression, StepDataEvaluator evaluator, ScenarioContext scenarioContext, Map<String, Object> localContext) {
+        if (evaluator == null) {
+            return toMilliSeconds(expression);
+        }
+        try {
+            ScenarioContextImpl mergedContext = new ScenarioContextImpl();
+            mergedContext.putAll(scenarioContext);
+            mergedContext.putAll(localContext);
```

Comment:
I would put this merge process out of this method and use directly a `Map<String, Object>` instead of a `ScenarioContextImpl`.
Besides, it will be executed once.